src/VisualStudio/TestUtilities/Remoting/RemotingWorkspace.cs

```diff
@@ -0,0 +1,47 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Diagnostics;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.VisualStudio.ComponentModelHost;
+using Microsoft.VisualStudio.LanguageServices;
+using Microsoft.VisualStudio.Shell;
+using RoslynProject = Microsoft.CodeAnalysis.Project;
+using RoslynWorkspace = Microsoft.CodeAnalysis.Workspace;
+
+namespace Roslyn.VisualStudio.Test.Utilities.Remoting
+{
+    internal class RemotingWorkspace
+    {
+        public static void AddReference(string projectName, string referenceAssemblyPath)
+        {
+            var metadataReference = MetadataReference.CreateFromFile(referenceAssemblyPath);
+            RoslynWorkspace workspace = GetWorkspace();
+            RoslynProject newProject = GetProject(workspace, projectName).AddMetadataReference(metadataReference);
+            workspace.TryApplyChanges(newProject.Solution);
+        }
+
+        public static void OpenDocument(string projectName, string documentName)
```

Comment:
This should be an instance method and `workspace` should be a field that is wrapped by the `WorkspaceWrapper` proxy.